File: 0415-Add-Strings/0415.js
// 415. 字符串相加
// 给定两个字符串形式的非负整数 num1 和num2 ，计算它们的和。

// 注意：

// num1 和num2 的长度都小于 5100.
// num1 和num2 都只包含数字 0 - 9.
// num1 和num2 都不包含任何前导零。
// 你不能使用任何內建 BigInteger 库， 也不能直接将输入的字符串转换为整数形式。
var addStrings = function(num1, num2) {
    let res = ''
    let i = num1.length - 1
    let j = num2.length - 1
    let sum = 0
    let flag = 0
    while (i >= 0 || j >= 0 || flag) {
        let n1 = +num1[i--] || 0
        let n2 = +num2[j--] || 0
        sum = n1 + n2 + flag
        if (sum >= 10) {
            flag = 1
            sum = sum - 10
        } else {
            flag = 0
        }
        res = sum + '' + res
    }
    return res;
}

File: 401-500/0415-Add-Strings/0415.go
// 415. 字符串相加
// 给定两个字符串形式的非负整数 num1 和num2 ，计算它们的和并同样以字符串形式返回。

// 你不能使用任何內建的用于处理大整数的库（比如 BigInteger）， 也不能直接将输入的字符串转换为整数形式。

// 示例 1：

// 输入：num1 = "11", num2 = "123"
// 输出："134"
// 示例 2：

// 输入：num1 = "456", num2 = "77"
// 输出："533"
// 示例 3：

// 输入：num1 = "0", num2 = "0"
// 输出："0"

// 提示：

// 1 <= num1.length, num2.length <= 10^4
// num1 和num2 都只包含数字 0-9
// num1 和num2 都不包含任何前导零

func addStrings(num1 string, num2 string) string {
	add := 0
	ans := ""
	for i, j := len(num1)-1, len(num2)-1; i >= 0 || j >= 0 || add != 0; i, j = i-1, j-1 {
		var x, y int
		if i >= 0 {
			x = int(num1[i] - '0')
		}
		if j >= 0 {
			y = int(num2[j] - '0')
		}
		result := x + y + add
		ans = strconv.Itoa(result%10) + ans
		add = result / 10
	}
	return ans
}